client/src/views/viewers/VideoViewer.vue

```diff
@@ -102,103 +104,45 @@ const cancelVolumeWatch = watch(
-//       { label: 'French', callback: (): Promise<void> => changeAudioLanguage(1) },
-//     ],
-//   },
-//   {
-//     label: 'Subtitles',
-//     icon: cog,
-//     children: [
-//       { label: 'None', callback: (): Promise<void> => changeSubtitlesLanguage(0), isActive: true },
-//       { label: 'English', callback: (): Promise<void> => changeSubtitlesLanguage(1) },
-//       { label: 'French', callback: (): Promise<void> => changeSubtitlesLanguage(2) },
-//     ],
-//   },
-// ]);
-
-// async function changePip(): Promise<void> {
-//   pipState.value = !pipState.value;
-//   dropdownItems.value[2].isActive = pipState.value;
-// }
-
-// async function changeLoop(): Promise<void> {
-//   loopState.value = !loopState.value;
-//   dropdownItems.value[3].isActive = loopState.value;
-// }
-
-// async function changeResolution(option: number): Promise<void> {
-//   dropdownItems.value[0].children?.forEach((item, index) => {
-//     item.isActive = option === index;
-//   });
-// }
-
-// async function changePlaybackSpeed(option: number): Promise<void> {
-//   dropdownItems.value[1].children?.forEach((item, index) => {
-//     item.isActive = option === index;
-//   });
-// }
-
-// async function changeAudioLanguage(option: number): Promise<void> {
-//   dropdownItems.value[4].children?.forEach((item, index) => {
-//     item.isActive = option === index;
-//   });
-// }
-
-// async function changeSubtitlesLanguage(option: number): Promise<void> {
-//   dropdownItems.value[5].children?.forEach((item, index) => {
-//     item.isActive = option === index;
-//   });
-// }
+const loopState = ref(false);
+const pipState = ref(false);
+const dropdownItems = ref<FileControlsDropdownItemContent[]>([
+  {
+    label: 'fileViewers.players.playback.title',
+    icon: timer,
+    children: [
+      { label: 'fileViewers.players.playback.speed.0', callback: (): Promise<void> => changePlaybackSpeed(0) },
+      { label: 'fileViewers.players.playback.speed.1', callback: (): Promise<void> => changePlaybackSpeed(1) },
+      { label: 'fileViewers.players.playback.speed.2', callback: (): Promise<void> => changePlaybackSpeed(2), isActive: true },
+      { label: 'fileViewers.players.playback.speed.3', callback: (): Promise<void> => changePlaybackSpeed(3) },
+      { label: 'fileViewers.players.playback.speed.4', callback: (): Promise<void> => changePlaybackSpeed(4) },
+    ],
+  },
+  {
+    label: 'fileViewers.players.pip',
+    image: PipIcon,
+    callback: togglePictureInPicture,
+    isActive: false,
+  },
+  {
+    label: 'fileViewers.players.loop',
+    icon: infinite,
+    callback: toggleLoop,
+    isActive: false,
+  },
+]);
 
 onMounted(async () => {
   src.value = URL.createObjectURL(new Blob([props.contentInfo.data], { type: props.contentInfo.mimeType }));
+  document.addEventListener('enterpictureinpicture', () => onTogglePictureInPicture(true), false);
```

Comment:
Is there a reason why you add it to the document instead of the video element?

If not, add those directly on the video element,
```
<video
  @enterpictureinpicture="onTogglePictureInPicture(true)"
/>
This way you don't have to remove the event listener, Vue will do it for you.
```